src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -888,9 +884,10 @@ public override void VisitTargetScopeSyntax(TargetScopeSyntax syntax)
                 {
                     FunctionFlags.ParameterDecorator => diagnosticBuilder.ExpectedParameterDeclarationAfterDecorator(),
                     FunctionFlags.VariableDecorator => diagnosticBuilder.ExpectedVariableDeclarationAfterDecorator(),
-                    FunctionFlags.ResoureDecorator => diagnosticBuilder.ExpectedResourceDeclarationAfterDecorator(),
+                    FunctionFlags.ResourceDecorator => diagnosticBuilder.ExpectedResourceDeclarationAfterDecorator(),
                     FunctionFlags.ModuleDecorator => diagnosticBuilder.ExpectedModuleDeclarationAfterDecorator(),
                     FunctionFlags.OutputDecorator => diagnosticBuilder.ExpectedOutputDeclarationAfterDecorator(),
+                    FunctionFlags.ResourceOrModuleDecorator => diagnosticBuilder.ExpectedResourceOrModuleDeclarationAfterDecorator(),

```

Comment:
>ResourceOrModuleDecorator [](start = 34, length = 25)

Doing this was easier than trying to come up with a good error message for all possible combinations.